solver/bboltcachestorage/storage.go

```diff
@@ -455,3 +451,55 @@ func isEmptyBucket(b *bolt.Bucket) bool {
 	k, _ := b.Cursor().First()
 	return k == nil
 }
+
+// safeOpenDB opens a bolt database and recovers from panic that
+// can be caused by a corrupted database file.
+func safeOpenDB(dbPath string) (db *bolt.DB, err error) {
+	defer fallbackOpenDB(dbPath, &db, &err)
```

Comment:
I think if we are already adding a fallback then it should cover all the possible cases where boltdb finds a corrupted library that it can't open. Fallback is cheap and can't happen for regular reasons. Don't want to miss out something and wait for another release cycle.

Reply:
I've modified this to also apply to any error that occurs when `os.Stat` otherwise returns without error. So if stat fails for some reason (generally the file does not exist or is not readable), then we'll perform the fallback.